### PR TITLE
Fission

### PR DESCRIPTION
**Pull request by alandolfi278**
tracks epochs and validation loss, and updates some pieces of outdated code

### CODE REVIEW

File: CycleDiffusion.ipynb [hunk 1]
```diff
@@ -41,7 +41,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "ckpt = torch.load('logs_gen/Line_sim/ckpt_0.000000_1000000.pt', map_location=torch.device(device))\n",
+    "ckpt = torch.load('logs_gen/GEN_FissionSim/ckpt_0.000000_1000000.pt', map_location=torch.device(device))\n",
```
Comment (mpkuchera) [inline CycleDiffusion.ipynb]: These should be general to the dataset. Maybe a variable for this?

File: data/Fission/Fission_exp_data_processing.ipynb [hunk 2]
```diff
@@ -350,10 +350,13 @@
     "            else:\n",
     "                color = palette[0]\n",
     "            ax.scatter3D(x,y,z, color = color, s = 1)\n",
-    "\n",
-    "        ax.set_xlabel('x')\n",
-    "        ax.set_ylabel('y')\n",
-    "        ax.set_zlabel('z')\n",
+    "        sc = ax.scatter(x, z, y, c=colors, cmap=plt.cool(), s=1)\n",
+    "        ax.set_xlabel('X')\n",
```
Comment (mpkuchera) [inline data/Fission/Fission_exp_data_processing.ipynb:354]: I prefer the lower case

File: train/trainDPM.sh [hunk 3]
```diff
@@ -7,4 +7,5 @@
 source /opt/conda/bin/activate dpm-pc-gen
 
 # train generator
-python -m train.train_gen # Add arguments like --max_iters, --lr, --dataset_path, etc.
\ No newline at end of file
+python -m train.train_gen --dataset_path data/Fission/fission_data/Fission_sim.npy --max_iters 1000000 --val_freq 100 --tag FissionSim --lr 1e-3 --end_lr 1e-6
```
Comment (mpkuchera) [inline train/trainDPM.sh]: old code is better for the repo. This is too specific

**Review by mpkuchera — CHANGES_REQUESTED** [review]
overall this looks nice. Keep in mind that we don't want "hardcoded" paths and flags in repo. 

What is the purpose of `data/Fission` folder? it seems that data will be in `fission_data/xxxx`. Is our folder organization clear?

File: CycleDiffusion.ipynb [hunk 4]
```diff
@@ -74,8 +76,8 @@
    "outputs": [],
    "source": [
     "# Load the data\n",
-    "sim_data = np.load('data/toy/line.npy')\n",
-    "exp_data = np.load('data/toy/line_noisy.npy')\n",
+    "sim_data = np.load('data/Fission/fission_data/Fission_sim.npy')\n",
```
Comment (mpkuchera) [inline CycleDiffusion.ipynb:130]: make a variable with data path then load that data

File: CycleDiffusion.ipynb [hunk 5]
```diff
@@ -348,40 +348,39 @@
     "    fig, axs = plt.subplots(1, 2, figsize=(10, 5), subplot_kw={'projection': '3d'})\n",
     "    \n",
     "    # Scatter plot for the first point cloud\n",
-    "    pc1 = axs[0].scatter(x1, y1, z1, s=2, cmap='cool', vmin=0, vmax=8000)\n",
+    "    pc1 = axs[0].scatter(x1, z1, y1, s=2, cmap='cool', vmin=0, vmax=8000)\n",
```
Comment (mpkuchera) [inline CycleDiffusion.ipynb]: why do we need `vmax=8000`? What is the actual max?

File: data/Fission/fission_data_vis/.ipynb_checkpoints/Fission_exp_data_processing-checkpoint.ipynb
Comment (mpkuchera) [inline data/Fission/fission_data_vis/.ipynb_checkpoints/Fission_exp_data_processing-checkpoint.ipynb:1]: remove this file from commits

File: data/Fission/fission_data_vis/.ipynb_checkpoints/visualize-checkpoint.ipynb
Comment (mpkuchera) [inline data/Fission/fission_data_vis/.ipynb_checkpoints/visualize-checkpoint.ipynb:1]: remove this file from git tracking

File: data/Fission/fission_data_vis/.ipynb_checkpoints/Fission_sim_data_processing-checkpoint.ipynb
Comment (mpkuchera) [inline data/Fission/fission_data_vis/.ipynb_checkpoints/Fission_sim_data_processing-checkpoint.ipynb:1]: remove from git tracking

File: data/Fission/fission_data_vis/.ipynb_checkpoints/Fission_exp_data_processing-checkpoint.ipynb
Comment (mpkuchera) [inline data/Fission/fission_data_vis/.ipynb_checkpoints/Fission_exp_data_processing-checkpoint.ipynb:1]: remove from git tracking

File: data/Fission/fission_data_vis/.ipynb_checkpoints/Sim_and_Exp_Vis-checkpoint.ipynb
Comment (mpkuchera) [inline data/Fission/fission_data_vis/.ipynb_checkpoints/Sim_and_Exp_Vis-checkpoint.ipynb:1]: remove from git tracking

File: data/Fission/fission_data_vis/Fission_exp_data_processing.ipynb [hunk 6]
```diff
@@ -0,0 +1,526 @@
...
+   "source": [
+    "# User-Desired Settings"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "48a9ba78-ea1d-4468-87da-34a4095d1934",
+   "metadata": {},
+   "source": [
+    "The isotope used in this experiment is Mg22."
```
Comment (mpkuchera) [inline data/Fission/fission_data_vis/Fission_exp_data_processing.ipynb]: the reaction studied in this experiment is 22Mg + alpha

File: data/Fission/fission_data_vis/visualize.ipynb
Comment (mpkuchera) [inline data/Fission/fission_data_vis/visualize.ipynb:1]: you have a lot of vis notebooks. so, the title of this one should be more clear

File: train/trainDPM.sh [hunk 7]
```diff
@@ -4,7 +4,9 @@
 #SBATCH --mem 64G
 #SBATCH --gpus 1
 
-source /opt/conda/bin/activate dpm-pc-gen
+# source /opt/conda/bin/activate dpm-pc-gen
 
+cd /home/DAVIDSON/<yournamehere>/attpc_unpaired_trans
```
Comment (mpkuchera) [inline train/trainDPM.sh]: why do we require this line? Maybe just `cd ../` is sufficient...

**Review by mpkuchera — CHANGES_REQUESTED** [review]
changes requested, see code comments

File: CycleDiffusion.ipynb
Comment (mpkuchera) [inline CycleDiffusion.ipynb:1]: You need to clear all outputs and re-commit this

File: data/Fission/fission_data_vis/Fission_sim_data_processing.ipynb [hunk 8]
```diff
@@ -209,8 +209,14 @@
     "        zs = cloud[:, 2]\n",
     "        colors = cloud[:, 3]\n",
     "\n",
-    "        sc = ax.scatter(xs, ys, zs, c=colors, cmap=plt.cool(), s = 1)\n",
+    "        sc = ax.scatter(xs, zs, ys, c=colors, cmap=plt.cool(), s=1)\n",
     "        ax.set_title(f'Point Cloud {i+1}')\n",
+    "        ax.set_xlabel('X')\n",
```
Comment (mpkuchera) [inline data/Fission/fission_data_vis/Fission_sim_data_processing.ipynb]: I thought we changed back to 'x' 'y' 'z' labels?

**Review by mpkuchera — CHANGES_REQUESTED** [review]

File: CycleDiffusion.ipynb
Comment (alandolfi278) [reply CycleDiffusion.ipynb:1]: Done

Comment (rramanujan) [inline CycleDiffusion.ipynb:1]: Adding comments here, since GitHub doesn't allow line-by-line comments on notebooks.

1. There is duplicated code for computing the minimum validation loss. Refactor this into a function that you call twice instead.
2. In `visualize_point_cloud`: what are the following lines doing?
```
c_min = c[c > 0].min() if (c > 0).any() else 1e-6
c = c + c_min * 0.1
```
Perhaps to make the plot look prettier? Add comments please! Similarly, there's a later section with the following code:
```
offset = 1e-2
c1 = c1 + offset
c2 = c2 + offset
```
Comments to explain what's going on please!

File: PhysicsExperiments.ipynb
Comment (rramanujan) [inline PhysicsExperiments.ipynb:1]: Clear outputs before committing notebooks. Do this for all notebooks.

File: data/Fission/fission_data_vis/.ipynb_checkpoints/Fission_exp_data_processing-checkpoint.ipynb
Comment (rramanujan) [inline data/Fission/fission_data_vis/.ipynb_checkpoints/Fission_exp_data_processing-checkpoint.ipynb:1]: Why is this folder under version control? Checkpoint folders are maintained in Jupyter like "autosave" files. These should not be under version control. Add the `.ipynb_checkpoints` subfolder to `.gitignore` and remove these from the repo.

File: data/Fission/fission_data_vis/.ipynb_checkpoints/Fission_sim_data_processing-checkpoint.ipynb
Comment (rramanujan) [inline data/Fission/fission_data_vis/.ipynb_checkpoints/Fission_sim_data_processing-checkpoint.ipynb:1]: Should not be in version control.

File: data/Fission/fission_data_vis/.ipynb_checkpoints/Sim_and_Exp_Vis-checkpoint.ipynb
Comment (rramanujan) [inline data/Fission/fission_data_vis/.ipynb_checkpoints/Sim_and_Exp_Vis-checkpoint.ipynb:1]: Should not be in version control

File: README.md [hunk 9]
```diff
@@ -11,14 +11,14 @@ sbatch setup_env.sh
 ```
 
 ## Datasets
-We provided data processing notebooks inside the [data](data/) folder. Sample toy datasets can be generated by the data processing notebook from the [data](data/) folder  
+We provided data processing notebooks inside the [data](data/) folder. Sample toy datasets can be generated by the data processing notebook from the [data](data/) folder. In the [data/Fission folder](data/Fission/), both the fission data and the fission visualization can be found. See the [data folder](data/)
```
Comment (rramanujan) [inline README.md]: Don't format directory paths on the cluster as though they are hyperlinks. Just supply the path; it should look like `/data/Fission` etc., not like a clickable link. Also, note the leading `/` in the path, since this is an absolute path and not a relative one.

File: data/Fission/fission_data_vis/Sim_and_Exp_Vis.ipynb
Comment (rramanujan) [inline data/Fission/fission_data_vis/Sim_and_Exp_Vis.ipynb:1]: Again, top-level comment because I can't create a line-level comment.
1. Use relative paths whenever possible, since absolute paths are tied to your specific account on the cluster and will not work for someone else. So for example, instead of saying:
```
data = np.load('/home/DAVIDSON/allandolfi/attpc_unpaired_trans/data/Fission/fission_data/Fission_sim.npy')
```
you should say:
```
data = np.load('../Fission_sim.npy')
```
(assuming I understand your folder structure correctly -- adjust as needed.)

File: data/Fission/fission_data_vis/visualize_sim_exp_npy_files.ipynb
Comment (rramanujan) [inline data/Fission/fission_data_vis/visualize_sim_exp_npy_files.ipynb:1]: Why is the plotting code duplicated three times? Write a function that you call three times instead, with different file name arguments.

File: train/trainDPM.sh [hunk 10]
```diff
@@ -4,7 +4,9 @@
 #SBATCH --mem 64G
 #SBATCH --gpus 1
 
-source /opt/conda/bin/activate dpm-pc-gen
+# source /opt/conda/bin/activate dpm-pc-gen
```
Comment (rramanujan) [inline train/trainDPM.sh]: Wait, why do you no longer activate your virtual environment before launching your script?

File: train/train_gen.py [hunk 11]
```diff
@@ -15,13 +15,25 @@
...
-itters = []
+import numpy as np
+
+def get_unique_filename(directory, base_filename):
+    base, ext = os.path.splitext(base_filename)
+    candidate = os.path.join(directory, base_filename)
+    i = 1
+    while os.path.exists(candidate):
+        candidate = os.path.join(directory, f"{base}{i}{ext}")
+        i += 1
```
Comment (rramanujan) [inline train/train_gen.py:26]: Ok, but just for future reference: a cleaner way to do this would be to just use something that changes everytime you run this script for `i` instead, instead of doing this "search" over what `i` values have already been used. For example, you could use the SLURM assigned job id. No action needed on this one.

File: train/train_gen.py
Comment (rramanujan) [inline train/train_gen.py:1]: The control flow in this file is really hard to follow. Some code is in functions, other is not, so it's difficult to understand where the code starts executing and where it goes from there. Can we refactor this so that _all_ code lives inside functions? And then there's a single `main` function that controls flow?

**Review by rramanujan — CHANGES_REQUESTED** [review]
